Sofa/Component/Engine/Select/src/sofa/component/engine/select/BaseROI.h

```diff
@@ -0,0 +1,172 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+#include <sofa/component/engine/select/config.h>
+
+#include <sofa/type/Vec.h>
+#include <sofa/core/DataEngine.h>
+#include <sofa/core/objectmodel/BaseObject.h>
+#include <sofa/core/behavior/MechanicalState.h>
+#include <sofa/core/topology/BaseMeshTopology.h>
+#include <sofa/defaulttype/RigidTypes.h>
+
+namespace sofa::component::engine::select
+{
+
+/**
+ * This interface defines ROI features.
+ */
+template <class DataTypes>
+class BaseROI : public core::DataEngine
+{
+public:
+    using Inherit = core::DataEngine;
+    SOFA_CLASS(SOFA_TEMPLATE(BaseROI,DataTypes), Inherit);
+
+    using VecCoord = typename DataTypes::VecCoord;
+    using Coord = typename DataTypes::Coord;
+    using Real = typename DataTypes::Real;
+    using CPos = typename DataTypes::CPos;
+
+    using PointID = core::topology::BaseMeshTopology::PointID;
```

Comment:
Open question: Shouldn't we do alias that does not rely on a templated type of the class outside of the class (eg: in an anonymous namespace or in a private namespace).